examples/AdvancedOperations/AddAppCampaign.php

```diff
@@ -0,0 +1,466 @@
+        //   1. you cannot override bid settings at the ad group level.
+        //   2. you cannot add ad group criteria.
+        $adGroup = new AdGroup([
+            'name' => new StringValue(['value' => 'Earth to Mars cruises ' . uniqid()]),
+            'status' => AdGroupStatus::ENABLED,
+            'campaign' => new StringValue(['value' => $campaignResourceName])
+        ]);
+
+        // Creates an ad group operation.
+        $adGroupOperation = new AdGroupOperation();
+        $adGroupOperation->setCreate($adGroup);
+
+        // Submits the ad group operation to add the ad group and prints the results.
+        $adGroupServiceClient = $googleAdsClient->getAdGroupServiceClient();
+        $response = $adGroupServiceClient->mutateAdGroups($customerId, [$adGroupOperation]);
+
+        $createdAdGroupResourceName = $response->getResults()[0]->getResourceName();
+        printf(
+            "Created an ad group with resource name: '%s'.%s",
+            $createdAdGroupResourceName,
+            PHP_EOL
+        );
+
+        return $createdAdGroupResourceName;
+    }
+
+    /**
+     * Creates an App ad for a given ad group.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param string $adGroupResourceName the resource name of the ad group to add the App ad to
+     */
+    private static function createAppAd(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $adGroupResourceName
+    ) {
+        // Creates an ad group ad.
+        $adGroupAd = new AdGroupAd([
+            'status' => AdGroupAdStatus::ENABLED,
+            'ad_group' => new StringValue(['value' => $adGroupResourceName]),
+            'ad' => new Ad([
+                // ad_data is a 'oneof' message so setting app_ad
+                // is mutually exclusive with ad data fields such as
+                // text_ad, gmail_ad, etc.
+                'app_ad' => new AppAdInfo([
+                    'headlines' => [
+                        new AdTextAsset([
+                            'text' => new StringValue(['value' => 'A cool puzzle game'])
+                        ]),
+                        new AdTextAsset([
+                            'text' => new StringValue(['value' => 'Remove connected blocks'])
+                        ])
+                    ],
+                    'descriptions' => [
+                        new AdTextAsset([
+                            'text' => new StringValue(['value' => '3 difficulty levels'])
+                        ]),
+                        new AdTextAsset([
+                            'text' => new StringValue(['value' => '4 colorful fun skins'])
+                        ])
+                    ]
+                    // Optional: You can set up to 20 image assets for your campaign.
+                    // 'images' => [INSERT_AD_IMAGE_ASSET_ID(s)_HERE]
```

Comment:
Is this correct? I believe the images field takes an image object that accepts the asset ID. I don't think you can just pass an array of IDs.

Reply:
Good catch. This should be [INSERT_AD_IMAGE_RESOURCE_NAME(s)_HERE](https://github.com/googleads/google-ads-python/blob/667ac6d79c6c70480e47a78b5feb045c968d99cb/examples/advanced_operations/add_app_campaign.py#L275)

Reply:
Fix now under review https://github.com/googleads/google-ads-php/pull/280